Automattic/wp-calypso
# State: Introduce Redux version of the ads settings store

#### Changes proposed in this Pull Request

This PR introduces the Redux version of the ads settings store, including:
* Action creators
* Data layer
* Reducer (with tests)
* Selectors (with tests)
* Query component

#### Testing instructions

* Checkout this branch.
* Verify all tests pass.
* If you want to test it locally, you can:
  * Use log `state.wordads.settings` in your browser's console.
  * Then type `dispatch( { type: 'WORDADS_SETTINGS_REQUEST', siteId: 123456 } )` in your browser console, where `123456` is the ID of a site that has WordAds enabled.
  * Verify that `state.wordads.settings` now contains the WordAds settings of your site.
  * You can also drop the query component and the `getWordadsSettings` selector.
* You can also use #39798 to completely test this.

Part of #18222.
